hermes-test-helper/src/main/java/pl/allegro/tech/hermes/test/helper/builder/TopicBuilder.java

```diff
@@ -61,6 +64,14 @@ public Topic build() {
         );
     }
 
+    public TopicWithSchema buildAsTopicWithSchema() {
```

Comment:
can we use proper methods from `TopicWithSchema` where necessary instead of extending this builder? it feels somehow out of place to have build take arguments and return different type. Optionally, add `String schema` field to builder class and add `withSchema` method, then builder semantics is kept and you can get rid of `buildAsTopicWithSchema(String schema)` and only keep `buildTopicWithSchema()`:

```
topic().withSchema(...).buildTopicWithSchema()
```